bundles/org.eclipse.swt/Eclipse SWT PI/win32/library/os_custom.c

```diff
@@ -63,7 +63,7 @@ BOOL Validate_AllowDarkModeForWindow(const BYTE* functionPtr)
 	 * an ATOM value of 0xA91E which is unlikely to change
 	 */
 
-#ifdef _M_X64
+#if defined(_M_X64) || defined(_M_ARM64)
```

Comment:
This and other similar changes are not exactly correct, because on aarch64 disasm can't match the expected bytes. If you're not going to implement it now, it would be more correct to change this to

```
#elif defined(_M_ARM64)
    // Not implemented yet
    return FALSE;
```

Reply:
I don't understand the above objection at all.

What is the meaning of "aarch64 disasm can't match the expected bytes"? What is "aarch64 disasm" to do with this change in the C code?

What does it mean by "If you're not going to implement it now"? implement what exactly? The modified code compiles successfully with the Microsoft C/C++ native compiler for Arm64, so what kind of "implement" is needed here? I'm quite confused, to be honest.

If you object to the use of those predefined macros in the C code, they are specifically meant to be used with the Microsoft C/C++ compiler, targeting the Windows OS, as documented [here](https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170). This C code is not for any other OS but the Windows OS, so naturally the code is specifically designed to be compiled by the dominant compiler in the Windows system (Microsoft Visual Studio Suite).

The `build.bat` command file was also designed for Microsoft Visual Studio compiler tools, and my changes are to continue the usage of the same tools, for generating the native binaries for the Windows on Arm64 platform.

If you object to the use of Microsoft C/C++ tools here for the Windows OS, then this would become a huge topic that this PR can't address.

Reply:
The code in question tests compiled code (machine instructions, which can be seen as assembly code, which I described as disasm for shorthand). It verifies that machine code in a Windows library matches what's expected. This is because the APIs are not documented, but we still need them, so we need to check if at least it's stil lthe API we expected.

Surely aarch64 machine code differs from x86_64 machine code. That's the whole point of a different architecture. Otherwise you could just run x86_64 native libraries on aarch64.

Therefore, even if the code "compiles" for you, it will no work. To fix that, you have to implement checks specific to aarch64.

If you're not implementing that now, then change the code as I suggested, so that it's more clear that it's not implemented and doesn't look like "it's good".

Reply:
Specifically, as a result of NOT implementing the new checks, Dark Theme will stop working properly (scrollbars will be light, etc).

Reply:
Yes, I can see what you meant now. Thanks for the explanation. And here's a screenshot of Eclipse SDK in dark mode (built myself and running on my WoA box):

![image](https://user-images.githubusercontent.com/49886739/192858654-cf977243-284f-47d4-b76c-b3edf0b2a216.png)

As it shows, the scrollbars are not dark, so the code blocks within the
```
#if defined(_M_X64) || defined(_M_ARM64)
```
don't work properly in Windows Arm64; it may also not work if it just does `return FALSE;` instead, as suggested (I'll rebuild it and check it out later.)

In the mean time, I'll withdraw this PR, as it's become clear now that it's not useful, because the build infrastructure for building Windows on Arm64 artifacts is not yet in place to support it.

Reply:
aarch64 disasm taken care of at https://github.com/eclipse-platform/eclipse.platform.swt/pull/1048